scrapi/harvesters/vivo.py

```diff
@@ -0,0 +1,238 @@
+                    record[sparql_map['name']] = self.get_array(uri, sparql_map)
+                if sparql_map['type'] == 'dict':
+                    record[sparql_map['name']] = self.get_dict(uri, sparql_map)
+            records.append(record)
+        return records
+
+    def get_total(self, start_date, end_date):
+        query_str = """PREFIX vivo:  <http://vivoweb.org/ontology/core#>
+                       PREFIX xsd:   <http://www.w3.org/2001/XMLSchema#>
+                       SELECT (COUNT(*) AS ?total)
+                       {{
+                            ?s vivo:dateTimeValue ?dateURI .
+                            ?dateURI vivo:dateTime ?date .
+                            FILTER (strdt(?date, xsd:date) >= "{}"^^xsd:date && strdt(?date, xsd:date) <= "{}"^^xsd:date)
+                        }}"""
+        query_str = query_str.format(start_date.isoformat(), end_date.isoformat())
+        self.sparql_wrapper.setQuery(query_str)
+        result = self.sparql_wrapper.query()
+        result = result.convert()
+        return int(result['results']['bindings'][0]['total']['value'])
+
+    def get_uris(self, start_date, end_date, limit, offset):
+        uris = []
+        query_str = """ PREFIX vivo:  <http://vivoweb.org/ontology/core#>
+                        PREFIX xsd:   <http://www.w3.org/2001/XMLSchema#>
+
+                        SELECT ?uri
+                        {{
+                            ?uri vivo:dateTimeValue ?dateUri .
+                            ?dateUri vivo:dateTime ?date .
+                            FILTER (strdt(?date, xsd:date) >= "{}"^^xsd:date && strdt(?date, xsd:date) <= "{}"^^xsd:date)
+                        }}  LIMIT {} OFFSET {}
+                    """
+        query_str = query_str.format(start_date.isoformat(), end_date.isoformat(), limit, offset)
+        self.sparql_wrapper.setQuery(query_str)
+        results = self.sparql_wrapper.query()
+        results = results.convert()
+        for result in results['results']['bindings']:
+            uris.append(result['uri']['value'])
+        return uris
+
+    def complete_authors(self, authors):
+        for author in authors:
+            pattern = """ <{}> obo:ARG_2000028 ?vcard .
+                          ?vcard vcard:hasEmail ?emailUri .
+                          ?emailUri vcard:email ?email ."""
+            email = self.get_string(author['sameAs'], {'name': 'email', 'pattern': pattern, 'type': 'string'})
+            if email:
+                author['email'] = email
+
+            pattern = """?role obo:RO_0000052 <{}> .
+                         ?role vivo:roleContributesTo ?organization .
+                         ?organization rdfs:label ?name .
+                         ?role vivo:dateTimeInterval ?interval .
+                        FILTER NOT EXISTS {{ ?interval vivo:end ?end }}"""
+            affiliation = self.get_dict(author['sameAs'], {'name': 'affiliation', 'fields': ['name'], 'pattern': pattern, 'type': 'dict'})
```

Comment:
Could add some newlines to these long strings? Something like
```python
affiliation = self.get_dict(
    author['sameAs'], 
    {
        'name': 'affiliation', 
        'fields': ['name'], 
        'pattern': pattern, 
        'type': 'dict'
    }
)
```
Just to make it easier to parse,.